extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php

```diff
@@ -50,4 +52,25 @@ private static function getAppreciationLink( $revision ) {
 			],
 			wfMessage( 'appreciation-text' )->escaped() );
 	}
+
+	public function sendMail( \Revision $revision ) {
+		global $wgSitename;
+
+		if ( !$revision ) {
+			return;
+		}
+		$editedPageTitle = $revision->getTitle();
+
+		$params = [
+			'buttonLink' =>  SpecialPage::getTitleFor( 'Community' )->getFullURL(),
+			'targetUser' => $revision->getUserText(),
+			'editedPageTitleText' => $editedPageTitle->getText(),
+			'editedWikiName' => $wgSitename,
```

Comment:
There is no strict policy. During the performance work globals were faster ( because the don't use the magic __call, __get etc ) and thus the recommendation at that time was to use globals.